File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums = set(nums)
        longest = 0
        for num in nums:
            if num - 1 in nums:
                continue

            cur_cnt = 0
            cur_num = num
            while cur_num in nums:
                cur_num += 1
                cur_cnt += 1
            
            longest = max(longest, cur_cnt)
        
        return longest
            

File: Data Structures & Algorithms/maximum-subarray/submission-2.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int max_sum = -1001;
        int cur_sum = 0;
        for (int num : nums) {
            cur_sum = max(cur_sum + num, num);
            max_sum = max(max_sum, cur_sum);
        }

        return max_sum;
    }
};
